lib/ramen/test.js

```diff
@@ -0,0 +1,152 @@
+        analytics.assert(!window.ramenSettings);
+        analytics.assert(!window.Ramen);
+        analytics.initialize();
+        analytics.page();
+        analytics.assert(!window.ramenSettings);
+        analytics.assert(!window.Ramen);
+      });
+
+      it('should call #load', function(){
+        analytics.initialize();
+        analytics.page();
+        analytics.called(ramen.load);
+      });
+    });
+  });
+
+  describe('loading', function(){
+    it('should load', function(done){
+      analytics.load(ramen, done);
+    });
+  });
+
+  describe('after loading', function(){
+    beforeEach(function(done){
+      analytics.once('ready', done);
+      analytics.initialize();
+    });
+
+    describe('#identify', function(){
+      beforeEach(function(){
+        analytics.stub(window.Ramen, 'go');
+      });
+
+      it('should not call Ramen.go if only id is passed', function(){
+        analytics.identify('id');
+        analytics.assert(!window.ramenSettings);
+        analytics.didNotCall(window.Ramen.go);
+
+        analytics.identify('id');
+        analytics.assert(!window.ramenSettings);
+        analytics.didNotCall(window.Ramen.go);
+      });
+
+      it('should call Ramen.go and set correct attributes if just email passed', function(){
+        var email = 'email@example.com';
+        analytics.identify('id', { email: email });
+        analytics.assert(window.ramenSettings.organization_id === '6389149');
+        analytics.assert(window.ramenSettings.user.id === 'id');
+        analytics.assert(window.ramenSettings.user.name === email);
+        analytics.assert(window.ramenSettings.user.email === email);
+        analytics.called(window.Ramen.go);
+      });
+
+      it('should call Ramen.go and set correct attributes if email & name passed', function(){
+        var email = 'email@example.com';
+        var name = 'ryan+segment@ramen.is';
+        analytics.identify('id', { email: email, name: name });
+        analytics.assert(window.ramenSettings.organization_id === '6389149');
+        analytics.assert(window.ramenSettings.user.id === 'id');
+        analytics.assert(window.ramenSettings.user.name === name);
+        analytics.assert(window.ramenSettings.user.email === email);
+        analytics.called(window.Ramen.go);
+      });
+
+      it('should pass along integration options', function(){
+        var email = 'email@example.com';
+        var name = 'ryan+segment@ramen.is';
+        var auth_hash = 'authy_hasy';
+        var timestamp = new Date() / 1000;
+        var custom_links = [{href: 'https://ramen.is/support', title: 'Hello'}];
+        var labels = ['use', 'ramen!'];
+        var environment = 'staging';
+        var logged_in_url = 'https://align.ramen.is/manage';
+        var unknown_future_opt = '11';
+        var unknown_future_user_opt = 'user 11';
+
+        analytics.identify('id', {email: email, name: name},
+          {
+            integrations: {
+              Ramen: {
```

Comment:
thanks for the detail.

So it looks like there's some modifications we want to make with regards to a couple of these settings.

`environment` sounds like a integration-level setting and not call-specific, so we should take this out from here

`auth_hash`- is there a reason why you need the user email, id and name? It's possible email and name might not be specified. could you send us some docs on how you instruct your users to implement this so we can make sure we document how this works. I'm assuming this would be similar to [Intercom](https://segment.com/docs/integrations/intercom/#secure-mode) but any clarity would help. 

`custom_links`: doesn’t sound like this is related to the identify call either, but is rather an integration-level setting when the widget is loaded, so this should be removed from here. 

`logged_in_url`: this also sounds like an integration-level setting and not call specific.

Sound good?






environment- sounds like a integration-level setting and not call-specific, so we’ll take this out.
auth_hash- should be generated in the integration code by those fields, shouldn’t force our users to generate this.

custom_links: doesn’t sound like this is related to the identify call either, but is rather an integration-level setting when the widget is loaded

labels: @nathan should we include this as a generic trait rather than as Ramen specific option?

logged_in_url: this also sounds like an integration-level setting.. 

timestamp/auth_hash_timestamp: not sure why we can’t use the timestamp added by identify call, and use that to generate the auth_hash.